### PR TITLE
Revamp `signature` module docs

### PR DESCRIPTION
**Pull request by Arcticae**
## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: snforge_std/src/signature.cairo [hunk 1]
```diff
@@ -4,20 +4,32 @@ mod stark_curve;
 
 #[derive(Copy, Drop)]
 struct KeyPair<SK, PK> {
+    /// A key that is used for signing the messages, it's 2 times shorter than the signature itself
```
Comment (drknzz) [inline snforge_std/src/signature.cairo:7]:
```suggestion
    /// A key that is used for signing the messages
```
Not sure it's a necessary information

File: snforge_std/src/signature.cairo [hunk 2]
```diff
@@ -4,20 +4,32 @@ mod stark_curve;
...
 trait SignerTrait<T, H, U> {
+    /// Signs given message hash
+    /// `self` - KeyPair used for signing
+    /// `message_hash` - input to sign bounded by the curve type (u256 for 256bit curves, felt252 for StarkCurve)
+    /// Returns the signature components (usually r,s tuple)
     fn sign(self: T, message_hash: H) -> U;
 }
 
 trait VerifierTrait<T, H, U> {
+    /// `self` - KeyPair used for signing
```
Comment (drknzz) [inline snforge_std/src/signature.cairo:29]:
```suggestion
    /// `self` - KeyPair used for verifying
```

**Review by drknzz — APPROVED** [review]
💯

File: snforge_std/src/signature.cairo (hunk 1 above)
Comment (Arcticae) [reply snforge_std/src/signature.cairo:7]: I wanted to flex, sorry

File: snforge_std/src/signature.cairo [hunk 3]
```diff
@@ -4,20 +4,32 @@ mod stark_curve;
...
 #[derive(Copy, Drop)]
 struct KeyPair<SK, PK> {
+    /// A key that is used for signing the messages, it's 2 times shorter than the signature itself
     secret_key: SK,
+    /// A (x, y) point on the elliptic curve used for verification of the signature
     public_key: PK,
 }
 
 trait KeyPairTrait<SK, PK> {
+    /// Generates the private and public keys using the built-in random generator
```
[the remark this answers is not in the document]
Comment (Arcticae) [reply snforge_std/src/signature.cairo:14]: I think this terminology is more common, is it not accurate?

**Pull request merged**